### PR TITLE
API token authentication support for VMware Cloud Director

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR adds support for API token authentication for VMware Cloud Director.

**Which issue(s) this PR fixes**:

Required for https://github.com/kubermatic/kubermatic/issues/11624

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
VMware Cloud Director now supports authentication using API Token
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
Still WIP and blocked by https://github.com/kubermatic/machine-controller/pull/1612

**Comment by kron4eg** [comment]
Looks good. But I think you also need to update the MC image version to reference the latest commit.

File: pkg/credentials/credentials.go [hunk 1]
```diff
@@ -598,5 +599,61 @@ func vmwareCloudDirectorValidationFunc(creds map[string]string) error {
...
+		}
+	}
+
+	// Username and password are required when using default credentials i.e. username and password.
+	if !(username && password) {
+		return fail.CredentialsError{
+			Op:       "validating",
+			Provider: "VMware Cloud Director",
+			Err: errors.Errorf(
+				"key %v or %v is required but isn't present",
```
Comment (xmudrii) [inline pkg/credentials/credentials.go:651]: Nit:
```suggestion
				"key %v and %v are required but not present",
```

File: examples/terraform/vmware-cloud-director/README.md [hunk 2]
```diff
@@ -20,27 +20,34 @@ Following environment variables or terraform variables can be used to authentica
...
 | VCD_ORG | vcd.org |
 | VCD_URL | vcd.url |
 
+To use access token instead of username and password:
+
+
+| Environment Variable | Terraform Variable |
+|------|---------|
+| VCD_AUTH_TYPE | vcd.auth_type |
+| VCD_API_TOKEN | vcd.api_token |
```
Comment (xmudrii) [inline examples/terraform/vmware-cloud-director/README.md:33]: One thing that confuses me is that:
- we call this API token in Terraform and KubeOne
- it's called refresh token in the CSI config

What's the official name out of those two?

**Review by xmudrii — COMMENTED** [review]
Just two nits.

File: pkg/credentials/credentials.go (hunk 1 above)
Comment (ahmedwaleedmalik) [reply pkg/credentials/credentials.go:651]: I think the original one is fine.

Comment (xmudrii) [reply pkg/credentials/credentials.go:651]: The original one implies that you need only one (`or`), but my understanding is that we need both, right?

File: examples/terraform/vmware-cloud-director/README.md (hunk 2 above)
Comment (xmudrii) [reply examples/terraform/vmware-cloud-director/README.md:33]: Okay, that makes sense. Thanks!

File: pkg/credentials/credentials.go (hunk 1 above)
Comment (ahmedwaleedmalik) [reply pkg/credentials/credentials.go]: I have committed whatever your suggestion was, should be good now.

**Review by xmudrii — APPROVED** [review]
/approve

**Pull request merged**